src/main/java/org/kohsuke/github/Requester.java

```diff
@@ -1023,57 +1039,36 @@ private InputStream wrapStream(InputStream in) throws IOException {
     /**
      * Handle API error by either throwing it or by returning normally to retry.
      */
-    void handleApiError(IOException e) throws IOException {
-        int responseCode;
-        try {
```

Comment:
This case is handled in the main `_fetch` method now.  We pass in the best known values and don't have to guard against bad state here.